internal/services/sources_service.go

```diff
@@ -47,3 +49,12 @@ func GetSource(w http.ResponseWriter, r *http.Request) {
 		renderError(w, r, payloads.NewRenderError(r.Context(), "show source", err))
 	}
 }
+
+func fetchARN(ctx context.Context, client sources.SourcesIntegration, sourceId string) (string, error) {
+	resp, err := client.ListSourceAuthenticationsWithResponse(ctx, sourceId, &sources.ListSourceAuthenticationsParams{}, AddIdentityHeader)
```

Comment:
we will need to filter by our `application_type` here right? otherwise it would return all auths for this source (we are sure this source has our auth as well, but if the source is superkey source, it may have auths for multiple apps, right?

Reply:
the resouce_id is not the appid of our application (provisioning). it is the application created for a source.

Reply:
the resource_id is not equal to the appId (provisioning).

Reply:
And @adiabramovitch updated me on the existence of : https://github.com/RedHatInsights/sources-api-go/blob/ce5b90a2c2cbf51612ff96016cb288c02a24bca4/dao/seeds/source_types.yml#L69

Reply:
I added the correct filtering after asking on sources channel